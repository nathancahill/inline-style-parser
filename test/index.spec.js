const css = require('css');
const inlineStyleParser = require('..');
const { cases, snapshots } = require('./data');

/**
 * @param {string} inlineStyle
 * @return {Array}
 */
function getDeclarations(inlineStyle) {
  const {
    stylesheet: { rules }
  } = css.parse(`a{${inlineStyle}}`);
  return rules[0].declarations;
}

/**
 * @param {Object[]} declarations
 * @return {Object[]}
 */
function removePosition(declarations) {
  return declarations.map(({ type, property, value }) => ({
    type,
    property,
    value
  }));
}

it.each(cases)('parses "%s"', style => {
  expect(removePosition(inlineStyleParser(style))).toEqual(
    removePosition(getDeclarations(style))
  );
});

it.each(snapshots)('%s', (_name, style) => {
  expect(inlineStyleParser(style)).toMatchSnapshot();
});
