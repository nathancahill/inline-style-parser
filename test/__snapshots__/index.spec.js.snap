// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses correctly when comment precedes colon 1`] = `
Array [
  Object {
    "position": Position {
      "end": Object {
        "column": 40,
        "line": 1,
      },
      "source": undefined,
      "start": Object {
        "column": 1,
        "line": 1,
      },
    },
    "property": "text-align",
    "type": "declaration",
    "value": "center",
  },
]
`;
